support/util/util_test.go

```diff
@@ -485,3 +489,44 @@ func TestSanitizeIgnitionPayload(t *testing.T) {
 		})
 	}
 }
+
+func TestGetMgmtClusterCPUArch(t *testing.T) {
+	fakeKubeClient := fakekubeclient.NewSimpleClientset()
+	fakeDiscovery, ok := fakeKubeClient.Discovery().(*fakediscovery.FakeDiscovery)
+
+	if !ok {
+		t.Fatalf("failed to convert FakeDiscovery")
+	}
+
+	// if you want to fake a specific version
+	fakeDiscovery.FakedServerVersion = &apiversion.Info{
+		Platform: "linux/amd64",
+	}
+
+	tests := []struct {
+		Name         string
+		expectedArch string
+		expectedErr  bool
+	}{
+		{
+			Name:         "Nominal use case",
+			expectedArch: "amd64",
+			expectedErr:  false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			g := NewWithT(t)
+			t.Setenv("FAKE_CLIENT", "true")
```

Comment:
why do you need to set this env variable? you decoupled the client now

Reply:
This was left behind while I was debugging the unit tests. This has been removed now.